mlir/lib/Dialect/Rock/utility/CMakeLists.txt

```diff
@@ -19,4 +19,14 @@ add_rocmlir_dialect_library(MLIRRockUtility
   MLIRSupport
   MLIRRockAnalysis
   MLIRMHAL
+)
+
+find_package(hip REQUIRED)
+# We do need both of these.  One is for linking, to get the HIP
+# functions.  The other is for compiling, to get -D__HIP_PLATFORM_AMD__
+target_link_libraries(MLIRRockUtility PUBLIC
+  hip::host hip::amdhip64
+)
+target_link_libraries(obj.MLIRRockUtility PUBLIC
+  hip::host hip::amdhip64
```

Comment:
This is strange that you need to use `target_link_libraries` twice. I think even if you use target_link_libraries with `MLIRRockUtility` it should work

Reply:
you can try doing 
`target_compile_definitions(MLIRRockUtility PRIVATE __HIP_PLATFORM_AMD__)`

Reply:
https://github.com/ROCm/rocm-examples/blob/f9d4e5e78325c36b319d91ec37c6410b2b6e12fb/HIP-Basic/device_query/CMakeLists.txt#L83

Reply:
> you can try doing `target_compile_definitions(MLIRRockUtility PRIVATE __HIP_PLATFORM_AMD__)`


@mirza-halilcevic  did you try this one ?